src/main/scala/latis/service/hapi/HapiService.scala

```diff
@@ -26,9 +29,46 @@ import latis.server.ServiceInterface
  */
 class HapiService(catalog: Catalog) extends ServiceInterface(catalog) {
 
+  // The catalog provided by LaTiS contains datasets that cannot be
+  // represented with HAPI, so we filter those datasets out.
+  //
+  // This is a hack to filter out datasets that will obviously not
+  // work. The long-term solution is to rework the HAPI service so
+  // LaTiS datasets are turned into HAPI datasets when possible, and
+  // the ability to construct a HAPI dataset serves as proof that the
+  // LaTiS dataset is compatible with HAPI.
+  //
+  // This checks that:
+  // - The first domain variable is time
+  // - The time has units and coverage
+  // - Each scalar in the range has units and a supported type
+  // - If the type of a scalar is string, its size is defined
+  private val filteredCatalog: Catalog = {
+    val covP: Metadata => Boolean = _.getProperty("coverage").isDefined
+    val typeP: Metadata => Boolean = md => md.getProperty("type").exists {
+      case "string" => md.getProperty("size").isDefined
+      case "double" => true
+      case "int"    => true
+      case _        => false
```

Comment:
This is a safe place to start but I suspect we'll want to expand this, at least to include smaller types.